.circleci/config.yml

```diff
@@ -199,21 +200,23 @@ jobs:
           event: fail
           template: basic_fail_1
           branch_pattern: main
-
-  clean_up_ecr:
-    executor: cloud-platform-executor
+  build_and_push:
+    parameters:
+      image_type:
+        type: string
+    executor: aws-ecr/default
     steps:
       - checkout
-      - setup_remote_docker
-      - run:
-          name: Delete old images from ECR repo
-          command: |
-            ./bin/delete_ecr_images
-      - slack/notify:
-          channel: eligibility-alerts-uat
-          event: fail
-          template: basic_fail_1
-
+      - aws-cli/setup:
+          role_arn: $ECR_ROLE_TO_ASSUME
+          region: $ECR_REGION
+      - run: |
+          aws ecr get-login-password --region $ECR_REGION | docker login --username AWS --password-stdin ${AWS_ECR_REGISTRY_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com
+      - aws-ecr/build-image:
+          push-image: true
+          tag: << parameters.image_type >>-${CIRCLE_SHA1}
```

Comment:
What could the `aws-ecr/build-and-push-image` job not handle, that we need to go back to rolling our own?
Can this concatenation not be done in the workflow definition, or is it something else?

Reply:
The issue, as I understand it, is that aws-ecr doesn't know how to use OIDC short-lived credentials, so we need to define a job that uses aws-cli to handle authentication for us. At least, that's my read of the [guidance](https://user-guide.cloud-platform.service.justice.gov.uk/documentation/deploying-an-app/container-repositories/deprecating-long-lived-credentials.html#migrating-to-short-lived-credentials-for-circleci)

Reply:
Oh, you're absolutely right! How disappointing! Right you are, then :)